mirrord/agent/src/main.rs

```diff
@@ -285,12 +290,20 @@ impl ClientConnectionHandler {
                         break;
                     }
                 },
-                message = self.tcp_sniffer_api.recv(), if sniffer_available => {
+                // use to prevent closing when sniffer is not available
+                // while avoiding possible busy loop (infinite iteration of None)
+                message = async {
+                    if let Some(ref mut sniffer_api) = self.tcp_sniffer_api {
+                        sniffer_api.recv().await
+                    } else {
```

Comment:
And does the compiler force us to have this else clause? Is it impossible to have just the then clause?

Reply:
I think it's good since it's an assertion